### DISCUSSION TITLE
Including custom modules for Arduino Portenta H7

### DISCUSSION THREAD

**Discussion by djesic** [micropython/micropython#13772] [STM32 / Pyboard] [answered]
Hi,

I've been using an ESP32 based board and have managed to add some `.py` files to the `modules/` directory when creating my firmware.
Now I'm trying to create a firmware for the Arduino Portenta H7 and I'm just unsure what the procedure is to achieve the same thing is, as the `stm32/` does not have a `modules/` directory.

Any help would be appreciated, thanks!

**Comment by robert-hh** [comment, marked as answer]
This is all controlled by the manifest file sequence. See https://docs.micropython.org/en/latest/reference/manifest.html
For the Portenta H7, it starts with the manifest.py file in the boards directory of the Portenta H7. If you look at the esp32/boards/manifest.py file, it includes a line `freeze("$(PORT_DIR)/modules")`. If you add that e.g. to `stm32/boards/manifest.py` or to the manifest.py file of the Portenta H7, then all files in a stm32/modules directory will be include. You can as well make your own manifest.py file. add there `include("$(BOARD_DIR)/manifest.py")` and use `make FROZEN_MANIFEST=<my_mainfest_file> ...`
